### PR TITLE
Add support for the D4 pool

### PR DESCRIPTION
**Pull request by alphastorm**

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -356,6 +437,18 @@ export const POOLS_MAP: PoolsMap = {
     poolTokens: ALETH_POOL_TOKENS,
     isSynthetic: true,
   },
+  [D4_POOL_NAME]: {
+    name: D4_POOL_NAME,
+    addresses: D4_SWAP_ADDRESSES,
+    lpToken: D4_SWAP_TOKEN,
+    poolTokens: D4_POOL_TOKENS,
+    isSynthetic: false,
```
Comment (hammeiam) [inline src/constants/index.ts:445]: kind of a bummer in hindsight

File: src/hooks/useApproveAndDeposit.ts [hunk 2]
```diff
@@ -123,14 +123,14 @@ export function useApproveAndDeposit(
       if (isFirstTransaction) {
         minToMint = BigNumber.from("0")
       } else {
-        if (poolName === ALETH_POOL_NAME) {
-          minToMint = await (swapContract as SwapFlashLoanNoWithdrawFee).calculateTokenAmount(
+        if (isLegacySwapABIPool(poolName)) {
```
Comment (hammeiam) [inline src/hooks/useApproveAndDeposit.ts:126]: nice

**Review by hammeiam — APPROVED** [review]
lfg

**Review by penandlim — APPROVED** [review]
Tested deposit, withdraw, and swaps in forked maininet!

**Review by sandrasong — APPROVED** [review]

**Pull request merged**